tree/20_cfe_basics/log_rudder.cf

```diff
@@ -899,11 +899,11 @@ bundle agent method_id_push(method_id)
 bundle agent method_id_pop()
 {
   vars:
+    !pass1::
       "stack_len"                int => length("method_id_unique.stack");
+      "method_id"             string => nth("method_id_unique.stack", "0");
```

Comment:
I think that method_id should be only the stack is not empty
so here
not_empty_stack.!pass2::
"method_id"             string => nth("method_id_unique.stack", "0");

and 
pass1.!pass2::
"not_empty_stack" expression => isgreaterthan("${stack_len}", "0");